fili-core/src/main/java/com/yahoo/bard/webservice/table/PhysicalTable.java

```diff
@@ -102,4 +126,35 @@
     default ZonedTimeGrain getTimeGrain() {
         return getSchema().getTimeGrain();
     }
+
+    /**
+     * Create a constrained copy of this table.
+     *
+     * @param constraint  The dataSourceConstraint which narrows the view of the underlying availability
+     *
+     * @return a constrained table whose availability and serialization are narrowed by this constraint
+     */
+    ConstrainedTable withConstraint(DataSourceConstraint constraint);
+
+    /**
+     * Filter a map of raw intervals to those included in a physical schema.
+     *
+     * @param rawIntervals  The map of name to {@link SimplifiedIntervalList}s as the source availability
+     * @param schema  The schema describing the columns of this table
+     *
+     * @return map of column to set of available intervals
+     */
+    static Map<Column, SimplifiedIntervalList> getAllAvailableIntervals(
```

Comment:
The 'Map<String, SIL>' is the interface of the Availability getAllAvailableIntervals() while 'Map<Colulm,SIL>' is the interface of Table.getAllAvailableIntervals().
This method, properly translates from one to the other and is used in construction so it's problematic for it to be non static.